src/App.jsx

```diff
@@ -13,46 +15,107 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const ALPHABETICAL = 'Alphabetical';
+const BY_LENGTH = 'ByLength';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortGoodsByType = (goods, sortType, isReversed) => {
+  let sortedGoods = [...goods];
+
+  switch (sortType) {
+    case ALPHABETICAL:
+      sortedGoods.sort((good1, good2) => (
+        good1.localeCompare(good2)
+      ));
+      break;
+
+    case BY_LENGTH:
+      sortedGoods.sort((good1, good2) => (
+        good1.length - good2.length
+      ));
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    sortedGoods = sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetGoods = () => {
+    setIsReversed(false);
+    setSortType(null);
```

Comment:
when declaring a state, the default value is the empty string
but now is `null`
both options are valid, but it is better to choose one